native/common/include/jp_floattype.h

```diff
@@ -62,22 +62,32 @@ class JPFloatType : public JPPrimitiveType
 	virtual JPPyObject  getArrayItem(JPJavaFrame& frame, jarray, jsize ndx) override;
 	virtual void        setArrayItem(JPJavaFrame& frame, jarray, jsize ndx, PyObject* val) override;
 
+	template <class T> static T assertRange(const T& l)
+	{
+		if (l < -3.40282346638528860e+38 || l > 3.40282346638528860e+38)
```

Comment:
I guess these values are coming from the Java itself and would most likely never change in the future. Before your proposed change, these limits where obtained by a reflection. Actually I do think that there are different limits in different Java versions/Virtual machine implementations for float. However in the documentation the value is computed like this (2-2**(-23) * 2**127. I'm not sure if your expression could cause rounding errors.

Reply:
All values for type constants are defined in the Java language specification and thus not necessary to look up.  When i was pushing things arround the lookup got lost.

In the case of float we are actually implementing the wrong behavior here as I discovered in the testing.  Unlike ints where overflow makes sense,  floats are supposed to overflow to infinity when the go out of range.  

When i went to test it I discovered that some paths hit thus overflow test and other paths mapped to infinity.  When i tried to make it more consistent more tests broke so we appear to expect different behaviors in different locations.  Looking at the language specification it appears mapping to infinity is correct.  Thus this code appears to be buggy.  

I am fine with either but making this check appear on the other paths is slower, harder to achieve, and no correct according to the language spec.  Thus i recommend silently mapping to inf on out of range.  If you would like it to raise overflow we need some clear rules on the check and not just arbitrary.  

Ints have similar issues on the arbitrary application of overflow.  We test on construction of a wrapper type but not on an array assignment.